File: controller/Model/Utilities/PlayerTypeToStringConverter.cs
using System;
using System.Collections.Generic;
using System.Text;
using UltraPlayerController.Model.Enumerations;
using UltraPlayerController.Model.Exceptions;
using Utilities.Pattern;
using Utilities.Convert;

namespace UltraPlayerController.Model.Utilities
{
    public class PlayerTypeToStringConverter: IConverter<PlayerType, string>
    {
        private string _audioPlayerTypeText;
        private string _midiPlayerTypeText;
        private string _singlePlayerTypeText;
        private string _multyPlayerTypeText;

	#region Constructors
        private PlayerTypeToStringConverter()
        {
            _audioPlayerTypeText = "aud";
            _midiPlayerTypeText = "mid";
            _singlePlayerTypeText = "svi";
            _multyPlayerTypeText = "mvi";
        }
        public static PlayerTypeToStringConverter Instance
        {
            get
            {
                return Singleton<PlayerTypeToStringConverter>.Instance;
            }
        }
	#endregion

        #region IConverter<PlayerType,string> Members

        public PlayerType Convert(string data)
        {
            throw new NotImplementedException();
        }

        public string Convert(PlayerType data)
        {
            string result = string.Empty;
            switch (data)
            {
                case PlayerType.SingleMedia:
                    result = _singlePlayerTypeText;
                    break;
                case PlayerType.MultiMedia:
                    result = _multyPlayerTypeText;
                    break;
                case PlayerType.Audio:
                    result = _audioPlayerTypeText;
                    break;
                case PlayerType.Midi:
                    result = _midiPlayerTypeText;
                    break;
                default:
                    break;
            }
            return result;
        }

        #endregion
    }
}

File: player/ultraPlayer/PlayerTypeToStringConverter.cpp
#include "PlayerTypeToStringConverter.h"

PlayerTypeToStringConverter* PlayerTypeToStringConverter::_instance = NULL;

PlayerTypeToStringConverter::PlayerTypeToStringConverter(void)
{
}

PlayerTypeToStringConverter::~PlayerTypeToStringConverter(void)
{
}

PlayerType::Type PlayerTypeToStringConverter::Convert( string data )
{
    PlayerType::Type result;

    if (data == "aud")
    {
        result = PlayerType::Audio;
    }
    else if (data == "mid")
    {
        result = PlayerType::Midi;
    }
    else if (data == "svi")
    {
        result = PlayerType::SingleVideo;
    }
    else if (data == "mvi")
    {
        result = PlayerType::MultiVideo;
    }
    else
    {
        throw ConverterException("data = " + data, "PlayerTypeToStringConverter::Convert");
    }

    return result;
}

std::string PlayerTypeToStringConverter::Convert( PlayerType::Type data )
{
    throw NotImplementedException("", "PlayerTypeToStringConverter::Convert");
}

PlayerTypeToStringConverter* PlayerTypeToStringConverter::Instance()
{
    if (!_instance)
    {
        _instance = new PlayerTypeToStringConverter();
    }
    return _instance;
}
